start.py

```diff
@@ -222,17 +222,30 @@ def start():
         "--project-directory",
         "docker",
     ]
+    
+
     for compose_file in compose_files:
         base_command.append("-f")
         base_command.append(compose_file)
     # we use try/catch to mimick docker-compose's behaviour of handling CTRL+C event
     try:
-        command = base_command + [args.docker_command] + unknown
-        env = os.environ.copy()
-        env["DOCKER_BUILDKIT"] = "1"
-        if args.debug_build:
-            env["BUILDKIT_PROGRESS"] = "plain"
-        subprocess.run(command, env=env)
+         try:
+                command = base_command + [args.docker_command] + unknown
+                env = os.environ.copy()
+                env["DOCKER_BUILDKIT"] = "1"
+                if args.debug_build:
+                        env["BUILDKIT_PROGRESS"] = "plain"
+                subprocess.run(command, env=env, check=True)
+        #for docker compose v2
+        except OSError :
+                command = ["docker","compose"] + base_command[1:] + [args.docker_command] + unknown
+                env = os.environ.copy()
+                env["DOCKER_BUILDKIT"] = "1"
+                if args.debug_build:
+                        env["BUILDKIT_PROGRESS"] = "plain"
+                subprocess.run(command, env=env, check=True)
```

Comment:
this is cool but you have copy/pasted the code. Could you please optimize it with an external function?

Reply:
Yes I'll do that